packages/core/src/services/evaluations/run.ts

```diff
@@ -101,52 +101,50 @@ export const runEvaluation = async (
     required: ['result', 'reason'],
   }
 
-  const chainResult = await runChain({
+  const providersMap = await buildProvidersMap({
+    workspaceId: evaluation.workspaceId,
+  })
+  const run = await runChain({
     workspace,
     chain,
     source: LogSources.Evaluation,
-    apikeys: await buildProviderApikeysMap({
-      workspaceId: evaluation.workspaceId,
-    }),
+    providersMap,
     configOverrides: {
       schema,
       output: 'object',
     },
   })
 
-  if (chainResult.error) return chainResult
-
-  // Call the new method to handle the promise
-  chainResult.value.response.then((response) =>
-    handleEvaluationResponse(response, documentUuid, evaluation, documentLog),
+  const response = run.response as Promise<ChainStepResponse<'object'>>
+  response.then((res) =>
```

Comment:
hmm there's where I display my promises lack of knowledge but shouldn't you need to return res here so that if anyone is listening to this promise they keep getting the response?

Reply:
`response` is included in `run` so anyone consuming it can attach their `.then`. You did this xD